### DISCUSSION TITLE
Android Web Apps - Save file not working!

### DISCUSSION THREAD

**Discussion by coder4589** [apache/cordova#460] [General]
Does anybody here knows how to instruct "Cordova" and\or Android "WebView" to work with the "Save()" function below?

```
<!DOCTYPE html>

<a href='data:text/plain;charset=utf-8,This is a Test' download="Test.txt"> Save - Download </a>

<br><br>

<input type="button" value="Save - Download"  onclick='Save("User Data.txt", "Name: John" + "\r\n\r\n" + "Age: 18" + "\r\n\r\n" + "Country: USA")'>

<script>

function Save(filename, text) {    //_________________________ v1.0 ______________________________

var element = document.createElement('a');
element.setAttribute('href', 'data:text/plain;charset=utf-8,' + encodeURIComponent(text));
element.setAttribute('download', filename);

element.style.display = 'none';
document.body.appendChild(element);

element.click();

document.body.removeChild(element);

}

</script>
```

**Comment by breautek** [comment]
Short answer is, I don't think it's possible.

Long answer:

Webviews don't have a default implementation on how to handle "saves", but natively it can be handled via implementing a [DownloadListener](https://developer.android.com/reference/android/webkit/WebView#setDownloadListener(android.webkit.DownloadListener)), however I don't think the Cordova webview does this, thus downloads will not work.

Plugins I don't believe have direct access to the webview (because they shouldn't assume it to be an `android.webkit.WebView`) so I don't think it can be extended by a plugin, short of of implementing a webview plugin itself. So perhaps a PR would be considered if it can display an android intent to let the user select where to download the file to.

A potential workaround is to use the [file plugin](https://cordova.apache.org/docs/en/12.x/reference/cordova-plugin-file/index.html) instead, with the caveat that the file plugin should really only be used for internal/private app storage. If your intent is to let the user choose the file location so that they can easily access it outside of your application, then the file plugin might not be sufficient.

**Reply by coder4589** [reply]
Hello @breautek 

Thanks for replying.

"Short answer is, I don't think it's possible."

This is really disappointing, the "Save()" function mentioned in the main post is so simple and elegant, works so well in web browsers and just to find out that it does not work in Android\Cordova Webviews.

I was in hope there could be a simple solution for this issue, but I was wrong.

Anyway, thanks again for replying.
